commit 9415b7449c70f5ea4a0209ddb89c2f5f392d3b4b
Author: Muralidhara M K <muralimk@amd.com>
Date:   Tue Jul 27 06:36:45 2021 -0500

    rasdaemon: ras-mc-ctl: Fix script to parse dimm sizes
    
    Removes trailing spaces at the end of a line from
    file location and fixes --layout option to parse dimm nodes
    to get the size of each dimm from ras-mc-ctl.
    
    Issue is reported https://github.com/mchehab/rasdaemon/issues/43
    Where '> ras-mc-ctl --layout' reports all 0s
    
    With this change the layout option prints the correct dimm sizes
    > sudo ras-mc-ctl --layout
              +-----------------------------------------------+
              |                      mc0                      |
              |  csrow0   |  csrow1   |  csrow2   |  csrow3   |
    ----------+-----------------------------------------------+
    ...
    channel7: |  16384 MB  |     0 MB  |     0 MB  |     0 MB |
    channel6: |  16384 MB  |     0 MB  |     0 MB  |     0 MB |
    ...
    ----------+-----------------------------------------------+
    
    Signed-off-by: Muralidhara M K <muralimk@amd.com>
    Signed-off-by: Naveen Krishna Chatradhi <nchatrad@amd.com>
    Cc: Yazen Ghannam <yazen.ghannam@amd.com>
    Signed-off-by: Mauro Carvalho Chehab <mchehab@kernel.org>
    Link: https://lkml.kernel.org/r/20210810183855.129076-1-nchatrad@amd.com/

diff --git a/util/ras-mc-ctl.in b/util/ras-mc-ctl.in
index 1e3aeb7..b22dd60 100755
--- a/util/ras-mc-ctl.in
+++ b/util/ras-mc-ctl.in
@@ -246,6 +246,7 @@ sub parse_dimm_nodes
     if (($file =~ /max_location$/)) {
         open IN, $file;
         my $location = <IN>;
+        $location =~ s/\s+$//;
         close IN;
         my @temp = split(/ /, $location);
 
@@ -288,6 +289,7 @@ sub parse_dimm_nodes
 
         open IN, $file;
         my $location = <IN>;
+        $location =~ s/\s+$//;
         close IN;
 
         my @pos;
